red/runtime/storage/localfilesystem/projects/Project.js

```diff
@@ -693,6 +698,31 @@ function createProject(user, metadata) {
     })
 }
 
+function deleteProject(user, name) {
+    return when.promise(function(resolve,reject) {
+        checkProjectExists(name).then(function() {
+            if (currentProject && currentProject.name === name) {
+                var e = new Error("NLS: Can't delete the active project");
+                e.code = "cannot_delete_active_project";
+                return reject(e);
+            }
+            else {
+                return deleteProjectDirectory(name).then(function() {
+                    var projects = settings.get('projects');
+                    delete projects.projects[name];
+                    return settings.set('projects', projects);
+                })
+                .then(function() {
+                    return resolve();
+                })
+                .catch(function(err) {
+                    return reject(err);
+                });
+            }
+        });
+    });
+}
+
```

Comment:
We don't need to rely on the `when` library as much as node 4.x supports the basic `Promise` object - I would like to avoid adding more uses of it when it is not needed.

Line 702: remove the `when.promise` line
Line 703: The function should just `return checkProjectExists(name)....`
Line 707: Without the `resolve/reject` functions provided by `when`, you can just `throw e` here to cause the Promise to be rejected
Line 715-720: these are not needed as the promise is returned from `deleteProjectDirectory